server/src/controllers/orderController.service.ts

```diff
@@ -39,6 +40,48 @@ export const ticketingWebhook = async (
   }
 };
 
+export const updateRefundStatus = async (
+  prisma: ExtendedPrismaClient,
+  paymentIntent: string,
+  refundStatus: string,
+) => {
+  let refundStatusState : state;
+  switch(refundStatus) {
+    case 'succeeded':
+      refundStatusState = state.completed;
+      break;
+    case 'pending':
+      refundStatusState = state.in_progress;
+      break;
+    default:
+      refundStatusState = state.failed;
+      break;
+  }
+  const order = await prisma.orders.findUnique({
+    where: {
+      payment_intent: paymentIntent,
+    },
+    select: {
+      orderid: true,
+    }
+  });
+
+  if (order === null) {
+    // there is no forseeable case where this should happen
+    console.error('received refund for order that does not exist')
+    return;
+  }
+
+  await prisma.refunds.updateMany({
+    where: {
+      orderid_fk: order.orderid
+    },
+    data: {
+      refund_status: refundStatusState,
+    },
+  });
+};
```

Comment:
It looks like this function will update all refunds associated with a specific order to the same status, currently the db schema allows for multiple refunds per order so it might be worth updating the status based on the refund intent. I am not sure if the charge.refunded event sends the refund object, but it looks like the charge.refund.updated event sends the refund object so an easy way to go about it might be to grab the id (refund intent)  from that object and update the refund in our db based on that. Another possible route would be to grab the refund intent for all refunds associated with a payment intent then use `stripe.refunds.retrieve` to get their current statuses.

Reply:
Another thing we might want to do is delete all refund items associated with any failed refunds so that the status of associated order items is correctly represented in the system.

Reply:
You are correct that charge.refunded does not send the refund object while the charge.refund.updated event does. And while across all my tests, I have always gotten both a charge.refunded event and a charge.refund.updated event for refunds, the [Stripe docs says that updated events only are sent for "selected payment methods", which I could not find a better description for.](https://docs.stripe.com/refunds#refund-webhook-events) As such, I implemented your second idea since that seemed more resilient.

Reply:
Awesome thanks!